src/engine/renderer/tr_image.cpp

```diff
@@ -151,313 +152,252 @@ R_ListImages_f
+	std::string height = "heigth";
+	std::string depth = "depth";
+	std::string mm = "mm";
+	std::string type = "type";
+	std::string format = "format";
+	std::string twrap = "wrap.t";
+	std::string swrap = "wrap.s";
+	std::string name = "name";
+
+	// Number sizes
+	size_t numLen = 5;
+	size_t widthLen = 5;
+	size_t heightLen = 5;
+	size_t depthLen = 5;
+	size_t mmLen = 3;
+	size_t typeLen = 4;
+	size_t formatLen = 4;
+	size_t twrapLen = 4;
+	size_t swrapLen = 4;
+
+	// Header number sizes
+	numLen = std::max( numLen, num.length() );
+	widthLen = std::max( widthLen, width.length() );
+	heightLen = std::max( heightLen, height.length() );
+	depthLen = std::max( depthLen, depth.length() );
+	mmLen = std::max( mmLen, mm.length() );
+	typeLen = std::max( typeLen, type.length() );
+	formatLen = std::max( formatLen, format.length() );
+	twrapLen = std::max( twrapLen, twrap.length() );
+	swrapLen = std::max( swrapLen, swrap.length() );
+
+	// Value sizes
+	for ( const auto& kv : imageTypeName )
+	{
+		typeLen = std::max( typeLen, kv.second.length() );
+	}
+	
+	for ( const auto& kv : imageFormatNameSize )
+	{
+		formatLen = std::max( formatLen, kv.second.first.length() );
+	}
+
+	for ( const auto& kv: wrapTypeName )
+	{
+		// 2 for the "t." and "s." prefix length.
+		twrapLen = std::max( twrapLen, 2 + kv.second.length() );
+		swrapLen = std::max( swrapLen, 2 + kv.second.length() );
+	}
+
+	std::string separator = " ";
+	std::stringstream lineStream;
+
+	// Print header
+	lineStream << std::left;
+	lineStream << std::setw(numLen) << num << separator;
+	lineStream << std::right;
+	lineStream << std::setw(widthLen) << width << separator;
+	lineStream << std::setw(heightLen) << height << separator;
+	lineStream << std::setw(depthLen) << depth << separator;
+	lineStream << std::left;
+	lineStream << std::setw(mmLen) << mm << separator;
+	lineStream << std::setw(typeLen) << type << separator;
+	lineStream << std::setw(formatLen) << format << separator;
+	lineStream << std::setw(twrapLen) << twrap << separator;
+	lineStream << std::setw(swrapLen) << swrap << separator;
+	lineStream << name;
+
+	std::string lineSeparator( lineStream.str().length(), '-' );
+
+	Log::CommandInteractionMessage( lineSeparator );
+	Log::CommandInteractionMessage( lineStream.str() );
+	Log::CommandInteractionMessage( lineSeparator );
+
+	int texels = 0;
+	int dataSize = 0;
+
+	for ( int i = 0; i < tr.images.currentElements; i++ )
+	{
+		image_t *image = (image_t*) Com_GrowListElement( &tr.images, i );
 
 		if ( filter && !Com_Filter( filter, image->name, true ) )
 		{
 			continue;
 		}
 
-		Com_sprintf( buffer, sizeof( buffer ), "%4i: %4i %4i  %s   ",
-		           i, image->uploadWidth, image->uploadHeight, yesno[ image->filterType == filterType_t::FT_DEFAULT ] );
-		out += buffer;
-		switch ( image->type )
-		{
-			case GL_TEXTURE_2D:
-				texels += image->uploadWidth * image->uploadHeight;
-				imageDataSize = image->uploadWidth * image->uploadHeight;
+		num = Str::Format( "%i", i );
```

Comment:
None of these `Str::Format( %i, ... )` are needed. You can do `ostream << number`.

Reply:
Done.